### PR TITLE
DRILL-7830: Add XML as Supported Format for HTTP Plugin

### PR DESCRIPTION
**Pull request by cgivre**
# [DRILL-7830](https://issues.apache.org/jira/browse/DRILL-7830): Add XML as Supported Format for HTTP Plugin

## Description
This PR extends the HTTP/REST plugin to accept XML as input. This PR reuses Drill's existing XML reader.

## Documentation
This PR allows the REST plugin to query REST APIs which return XML data.  The REST plugin currently has an option called `inputType`.  The available options are `xml`, `json`, `csv`.   

This PR also adds an additional configuration option of `xmlDataLevel` which sets the starting nesting level for the XML data.  This helps reduce the amount of unhelpful nesting in the data.  You can view the [documentation for the XML plugin here](https://github.com/apache/drill/tree/master/contrib/format-xml).

## Testing
Added additional unit test.

### CODE REVIEW

File: contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpApiConfig.java
Comment (luocooong) [inline contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpApiConfig.java:135]: It is recommended to define "json" as a final static variable.

File: contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpScanBatchCreator.java
Comment (luocooong) [inline contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpScanBatchCreator.java:102]: Could 'csv' and 'xml' reuse the final static variables recommended above?

**Review by luocooong — APPROVED** [review]

File: contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpApiConfig.java
Comment (cgivre) [reply contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpApiConfig.java:135]: Fixed

File: contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpScanBatchCreator.java
Comment (cgivre) [reply contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpScanBatchCreator.java:102]: Fixed

**Comment by cgivre** [comment]
Thanks @luocooong for the quick review!  I addressed your changes and squashed commits.   Will wait until tomorrow to merge.
Thanks again!

@dzamo 
There are some minor updates to the docs included in this PR.  I updated the `README` for the HTTP storage plugin to reflect that this can now read XML.

**Comment by dzamo** [comment]
Docs updated and will reflect after the next site build.

**Pull request merged**
